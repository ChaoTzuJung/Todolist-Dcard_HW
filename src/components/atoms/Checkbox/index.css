.label {
	display: flex;
	align-items: center;
	position: relative;
	width: 100%;
	height: 30px;
	padding-left: 40px;
	font-size: 24px;
	font-weight: 500;
	line-height: 18px;
	letter-spacing: 0.5px;
	cursor: pointer;
	color: var(--black);

	& > .checkbox {
		display: flex;
		position: absolute;
		justify-content: center;
		align-items: center;
		left: 0;
		width: 24px;
		height: 24px;
		background-color: var(--white);
		border: 1px solid var(--white);
		border-radius: 2px;
		transition: all 0.2s;

		& > i {
			font-size: 18px;
			color: var(--white);
			opacity: 0;
		}
	}

	&.small {
		padding-left: calc(16px + 12px);
		line-height: 16px;
		font-size: 12px;

		& > .checkbox {
			width: 16px;
			height: 16px;
		}
	}

	& > input:checked ~ .checkbox {
		background: var(--blue);
		border: 1px solid var(--blue);

		& > i {
			opacity: 1;
		}
	}
}

input[type='checkbox'] {
	position: absolute;
	opacity: 0;
	cursor: pointer;
	width: 0;
	height: 0;
}
