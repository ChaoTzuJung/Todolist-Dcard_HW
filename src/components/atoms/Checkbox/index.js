import React, { Component } from 'react';
import classnames from 'classnames';

import Icon from 'components/atoms/Icon';

import styles from './index.css';

class Checkbox extends Component {
	constructor(props) {
		super(props);
		this.handleChange = this.handleChange.bind(this);
	}

	handleChange(evt) {
		const { onChange = () => {} } = this.props;

		onChange(evt.target.value, evt.target.checked);
	}

	render() {
		const { name, value, content, className, small, onChange: _, ...other } = this.props;
		return (
			<label
				htmlFor={name}
				className={classnames(styles.label, className, small ? styles.small : '')}
			>
				{content}
				<input
					type="checkbox"
					id={name}
					name={name}
					value={value}
					onChange={this.handleChange}
					{...other}
				/>
				<span className={styles.checkbox}>
					<Icon>check</Icon>
				</span>
			</label>
		);
	}
}

export default Checkbox;
