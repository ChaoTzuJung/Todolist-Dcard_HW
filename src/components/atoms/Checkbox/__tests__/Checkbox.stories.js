import React from 'react';
import { withKnobs, boolean } from '@storybook/addon-knobs/react';
import { storiesOf } from '@storybook/react';

import Checkbox from 'components/atoms/Checkbox';

const stories = storiesOf('Commons|atoms/Checkbox', module);

stories.addDecorator(withKnobs);

stories.add('__interactive', () => (
	<Checkbox name="checkbox1" value="1" content="可以養寵物" small={boolean('small')} />
));
