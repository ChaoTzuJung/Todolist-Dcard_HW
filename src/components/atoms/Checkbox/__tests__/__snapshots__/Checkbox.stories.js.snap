// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`__interactive 1`] = `
<Checkbox
  content="可以養寵物"
  name="checkbox1"
  value="1"
>
  <label
    className="label"
    htmlFor="checkbox1"
  >
    可以養寵物
    <input
      id="checkbox1"
      name="checkbox1"
      onChange={[Function]}
      type="checkbox"
      value="1"
    />
    <span
      className="checkbox"
    >
      <Icon>
        <i
          className="icon material-icons"
        >
          check
        </i>
      </Icon>
    </span>
  </label>
</Checkbox>
`;
